Editor update particle-system inspector

chichinohaha opened the pull request:
Re: cocos-creator/3d-tasks#

Changelog:
- add cull category
    ![image](https://user-images.githubusercontent.com/59186205/135235776-318c55e4-fdc0-4c25-ac5a-ecbd0ba46990.png)
    ![image](https://user-images.githubusercontent.com/59186205/135235910-478a4723-9ba9-4aa4-8e8f-fee9f2582bd8.png)

 
fixed https://github.com/cocos-creator/3d-tasks/issues/9262
reply https://github.com/cocos-creator/3d-tasks/issues/6849

TheArtOfGameDesign committed:
change custom property display order

TheArtOfGameDesign committed:
change custom property display order

TheArtOfGameDesign committed:
fix display name of startsize x

TheArtOfGameDesign committed:
fix display name of startsize x

TheArtOfGameDesign committed:
add margin-left

TheArtOfGameDesign committed:
add margin-left

TheArtOfGameDesign committed:
update particle-system-inspector

- use updateCustomPropElements to update custom element
- add cull category

TheArtOfGameDesign committed:
update particle-system-inspector

- use updateCustomPropElements to update custom element
- add cull category

arsen2010 commented on editor/inspector/utils/prop.js:
```diff
@@ -30,25 +30,57 @@ exports.sortProp = function(propMap) {
+            if (!node) {
+                node = document.createElement('ui-prop');
+                node.setAttribute('type', 'dump');
+                prop.dump.displayOrder = prop.dump.displayOrder === undefined ? 0 : Number(prop.dump.displayOrder);
```
set default displayOrder value should deal in ‘sortProp’ method

chichinohaha replied:
done

TheArtOfGameDesign committed:
update sortProp and add particle-system show bounds button

update sortProp
- item may be null

add particle-system show bounds button
-

TheArtOfGameDesign committed:
update sortProp and add particle-system show bounds button

update sortProp
- item may be null

add particle-system show bounds button
-

TheArtOfGameDesign committed:
add resetBounds button

TheArtOfGameDesign committed:
add resetBounds button

TheArtOfGameDesign committed:
fix dispatch bug

TheArtOfGameDesign committed:
fix dispatch bug

TheArtOfGameDesign committed:
Change the content of the panel, fix the multi-selection situation, remove the inputflag

Change panel content
-The ui-prop that uses this tag sets the disabled of the content-slot to the marked dump.value

-When culling is not turned on, disable the elements in the enableCulling column

Fix multiple selection
-showBounds adapts to the case of multiple selections, when changing, it sends message for each component
-The Regenerate bounding box button will be set to disabled according to the selected component enableCulling
-Fix the ui-checkbox missing invalid judgement in ui-section header
-Fix the problem that ui-prop is not set dump.values to each component when multiple-selection

Remove inputflag
-No panel uses this mark, remove the related code

TheArtOfGameDesign committed:
Change the content of the panel, fix the multi-selection situation, remove the inputflag

Change panel content
-The ui-prop that uses this tag sets the disabled of the content-slot to the marked dump.value

-When culling is not turned on, disable the elements in the enableCulling column

Fix multiple selection
-showBounds adapts to the case of multiple selections, when changing, it sends message for each component
-The Regenerate bounding box button will be set to disabled according to the selected component enableCulling
-Fix the ui-checkbox missing invalid judgement in ui-section header
-Fix the problem that ui-prop is not set dump.values to each component when multiple-selection

Remove inputflag
-No panel uses this mark, remove the related code

TheArtOfGameDesign committed:
replace utils function ,update  "sortProp" function

replace utils function
-  "getCustomPropElements" -> "updateCustomPropElements"
this function is use to render custom node

update  "sortProp" function
- item may be null

TheArtOfGameDesign committed:
Update particle-system.js

chichinohaha committed:
replace utils function ,update "sortProp" function (#9425)

* replace utils function ,update  "sortProp" function

replace utils function
-  "getCustomPropElements" -> "updateCustomPropElements"
this function is use to render custom node

update  "sortProp" function
- item may be null

* Update particle-system.js

yanOO1497 committed:
[3.4.0] Add UI_GPU_DRIVEN flag in cc.config (#9367)

* update flags

* update flags

TheArtOfGameDesign committed:
Merge branch 'EDITOR-3d-particle-Adjust-the-display-order' of https://github.com/chichinohaha/engine into EDITOR-3d-particle-Adjust-the-display-order

chichinohaha commented:
https://github.com/cocos-creator/engine/pull/9426
